discovery/gossiper.go

```diff
@@ -1588,7 +1590,7 @@ func (d *AuthenticatedGossiper) processNetworkAnnouncement(
 
 		nMsg.err <- nil
 		// TODO(roasbeef): get rid of the above
-		return announcements
+		return announcements, true
```

Comment:
Hmm, trying to think of a slightly less error prone way to communicate this information...

Reply:
If `processNetworkAnnouncement` would return the error instead of sending it over the channel, we'd be able to inspect and determine this as it's done in the router.

Reply:
make `processNetworkAnnouncement` return the error and have the `networkHandler` send it on the `nMsg.err` channel?

Reply:
Not a bad idea, this would be an intermediate step compared to what I suggested above re a generic error send higher-order function.